src/Pages/Music/Artists.elm

```diff
@@ -88,10 +128,52 @@ view model =
     { title = "Music.Artists"
     , body =
         [ row [ Element.height fill, Element.width fill ]
-            [ Components.VerticalNavMusic.view model.route
+            [ column [ Element.height fill, Element.width fill ]
+                [ Components.VerticalNavMusic.view model.route
+                , column [ Element.height fill, Element.width fill, paddingXY 20 30, Background.color Colors.sidebar, spacingXY 0 15 ]
+                    [ Element.text "SORT"
+                    , sortButton model.currentButton (Title Asc) "Title " TitleButtonMsg
+                    , sortButton model.currentButton (Random Asc) "Random " RandomButtonMsg
+                    ]
+                ]
             , column [ Element.height fill, Element.width (fillPortion 6), paddingXY 0 0, spacingXY 5 7, Background.color Colors.background ]
                 [ Components.SectionHeader.viewArtists model.artist_list
                 ]
             ]
         ]
     }
+
+
+sortButton : ArtistSort -> ArtistSort -> String -> msg -> Element msg
+sortButton currentButton button name buttonMsg =
+    let
+        isCurrentButton =
+            case ( currentButton, button ) of
+                ( Title _, Title _ ) ->
+                    ( True, Title )
+
+                ( Random _, Random _ ) ->
+                    ( True, Random )
+
+                _ ->
+                    ( False, Random )
+    in
+    Input.button [ paddingXY 10 0 ]
+        { onPress = Just buttonMsg
+        , label =
+            currentButtonText currentButton name isCurrentButton
+        }
+
+
+currentButtonText : ArtistSort -> String -> ( Bool, SortDirection -> ArtistSort ) -> Element msg
```

Comment:
Seeing this, we might actually want to move the button look to one shared place, as this should look and work the same way everywhere?

Reply:
Hi @razzeee , 
I don’t think we can move the `currentButtonText` to one shared place because  ,while it appears and functions the same, it takes in different types in the parameter according to that page (AlbumSort,ArtistSort..). If we want it to be at one shared place then we will have to create a common type rather than multiple types like AlbumSort.

Reply:
Yeah, that will have some bad implications, you're right.